FluentTerminal.SystemTray/Utilities.cs

```diff
@@ -569,44 +569,58 @@ internal static void SaveFile(string path, string content)
             }
         }
 
-        internal static void MuteTerminal(bool mute)
+        private static bool MuteConhost(bool mute, int attemptCount)
         {
-            bool? deviceMuted = VolumeControl.GetDefaultAudioEndpointMute();
-            if (deviceMuted != true)
-            {
-                VolumeControl.SetDefaultAudioEndpointMute(true);
-            }
-
-            Process cmdProcess = new Process();
-            cmdProcess.StartInfo.FileName = "cmd.exe";
-            cmdProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
-            cmdProcess.StartInfo.Arguments = "/k \x07";
-            cmdProcess.Start();
-
-            bool wasConhostMuted = false;
+            bool audioSessionFound = false;
             int attempt = 0;
-            while (wasConhostMuted == false && attempt++ < 2 && !cmdProcess.HasExited)
+            while (audioSessionFound == false && attempt++ < attemptCount)
             {
-                cmdProcess.WaitForExit(1500);
-
                 foreach (Process conhost in Process.GetProcessesByName("conhost"))
                 {
                     try
                     {
-                        bool? isMuted = null;
-                        isMuted = VolumeControl.GetAudioSessionMute(conhost.Id);
-                        if (isMuted != null && isMuted != mute)
+                        bool? isAudioSessionMuted = null;
+                        isAudioSessionMuted = VolumeControl.GetAudioSessionMute(conhost.Id);
+                        if (isAudioSessionMuted != null && isAudioSessionMuted != mute)
                         {
                             VolumeControl.SetAudioSessionMute(conhost.Id, mute);
-                            wasConhostMuted = true;
                             Logger.Instance.Debug($"Mute state for conhost process with id={conhost.Id} was set to {mute}.");
                         }
+
+                        if (isAudioSessionMuted != null)
+                        {
+                            audioSessionFound = true;
+                        }
                     }
                     catch (Exception) { }
                 }
             }
+            return audioSessionFound;
+        }
+
+        internal static void MuteTerminal(bool mute)
+        {
+            bool audioSessionFound = MuteConhost(mute, 1);
+
+            if (audioSessionFound == true)
+                return;
+
+            bool? deviceMuted = VolumeControl.GetDefaultAudioEndpointMute();
+            if (deviceMuted != true)
+            {
+                VolumeControl.SetDefaultAudioEndpointMute(true);

```

Comment:
@mjs what do you think if to remove muting of audio device completely and let it play bell sound? So far it's 3-4 reports from different people to explain why the sound suddenly disappears.

Reply:
Yeah, I saw those reports. 

If the bell plays then people will probably complain about that... it might be less confusing than muting I guess.

[Terminus](https://www.termius.com/) lets the user disable the terminal bell (or use a visual bell) and they use xtermjs too. It might be good to dig into how they do it.

Reply:
@mjs will check Terminus. Initially, I have seen that Fluent passes some option to xterm to configure bell, but I haven't seen any difference in real behavior.

Reply:
https://github.com/felixse/FluentTerminal/blob/96ac5cd49fb1f4e5f6c79e7932fe4c029f0f15e8/FluentTerminal.Client/src/index.ts#L50

Reply:
Never mind about Terminus. I've just played around with it and its terminal beep suppression doesn't work.